### ISSUE TITLE
Incorrect CA1822 for members that implement an interface

### ISSUE THREAD

**Issue by marek-safar** [dotnet/roslyn-analyzers#5911]
### Analyzer

**Diagnostic ID**: [CA1822](https://docs.microsoft.com/dotnet/fundamentals/code-analysis/quality-rules/ca1822): `Member does not access instance data and can be marked as static`

### Analyzer source

**SDK**: [Built-in CA analyzers in .NET 5 SDK or later](https://docs.microsoft.com/dotnet/fundamentals/productivity/code-analysis)

**Version**: [SDK 6.0.100](https://dotnet.microsoft.com/download/dotnet/6.0)

### Describe the bug

CA1822 diagnostic is reported for a method that cannot be made static

### Steps To Reproduce

```c#
using System;
using System.Collections.Generic;

internal class C : A, IDisposable
{
    public static void Main ()
    {
        IDisposable ia = new C ();
        ia.Dispose ();
    }
}

internal class A
{
    public void Dispose ()
    {
    }
}
```

### Expected behavior

No CA1822

### Actual behavior

CA1822: Member 'Dispose' does not access instance data and can be marked as static

**Comment by mavasani** [comment]
> I know this is a simplified example but having A implementing the dispose pattern without the interface doesn't seem like a good pattern (having the interface on A would avoid the issue).

I agree - If `A.Dispose()` is implementing the dispose method (whether of itself for any of it's subtype), it should be marked as implementing `IDisposable`. This is by-design.
